apps/server/src/modules/account/services/account-db.service.spec.ts

```diff
@@ -176,350 +78,861 @@ describe('AccountDbService', () => {
+					userId: accountToSave.userId,
+					systemId: accountToSave.systemId,
+					createdAt: accountToSave.createdAt,
+					updatedAt: accountToSave.updatedAt,
+				});
+			});
 		});
 
-		it('should not change password if password is empty while editing an existing account', async () => {
-			const spy = jest.spyOn(accountRepo, 'save');
-			const dto = {
-				id: mockTeacherAccount.id,
-				// username: 'john.doe@domain.tld',
-				password: undefined,
-			} as AccountDto;
-			(accountRepo.findById as jest.Mock).mockClear();
-			(accountRepo.save as jest.Mock).mockClear();
-			await expect(accountService.save(dto)).resolves.not.toThrow();
-			expect(accountRepo.findById).toHaveBeenCalled();
-			expect(spy).toHaveBeenCalledWith(
-				expect.objectContaining({
+		describe("when account's system undefined", () => {
+			const setup = () => {
+				const mockUserWithoutAccount = userFactory.buildWithId();
+
+				const accountToSave: Account = {
+					createdAt: new Date(),
+					updatedAt: new Date(),
+					username: 'asdf@asdf.de',
+					userId: mockUserWithoutAccount.id,
 					password: defaultPassword,
-				})
-			);
+				} as Account;
+				(accountRepo.findById as jest.Mock).mockClear();
+				(accountRepo.save as jest.Mock).mockClear();
+
+				accountRepo.save.mockResolvedValue(
+					new Account({
+						id: new ObjectId().toHexString(),
+						username: accountToSave.username,
+						userId: accountToSave.userId,
+						createdAt: accountToSave.createdAt,
+						updatedAt: accountToSave.updatedAt,
+					})
+				);
+
+				return { accountToSave };
+			};
+
+			it('should keep undefined on save', async () => {
+				const { accountToSave } = setup();
+
+				const ret = await accountService.save(accountToSave);
+				expect(ret).toBeDefined();
+				expect(accountRepo.save).toBeCalledWith(expect.objectContaining({ systemId: undefined }));
+			});
+		});
+
+		describe('when save account', () => {
+			const setup = () => {
+				const mockUserWithoutAccount = userFactory.buildWithId();
+
+				const accountToSave = {
+					createdAt: new Date(),
+					updatedAt: new Date(),
+					username: 'asdf@asdf.de',
+					userId: mockUserWithoutAccount.id,
+					systemId: '012345678912',
+					password: defaultPassword,
+				} as Account;
+				(accountRepo.findById as jest.Mock).mockClear();
+				(accountRepo.save as jest.Mock).mockClear();
+
+				accountRepo.save.mockResolvedValue(
+					new Account({
+						id: new ObjectId().toHexString(),
+						username: accountToSave.username,
+						userId: accountToSave.userId,
+						createdAt: accountToSave.createdAt,
+						updatedAt: accountToSave.updatedAt,
+					})
+				);
+
+				return { accountToSave };
+			};
+
+			it('should encrypt password', async () => {
+				const { accountToSave } = setup();
+
+				const ret = await accountService.save(accountToSave);
+				expect(ret).toBeDefined();
+
+				expect(accountRepo.save).toBeCalledWith(
+					// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
```

Comment:
// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment should not needed anymore, place look to newer implementations of the objectContaining